### PR TITLE
EAR 2033 and 2035 - section page description redesign

### PR DESCRIPTION
**Pull request by farres1**
> ### BEFORE MAKING YOUR PR
>
> Please ensure:
>
> - There are no linting errors, all tests must pass
> - PR is named after JIRA ticket number e.g. EAR-###
> - **Accesibility** checks are completed:
>   - Elements have discernible and consistent focus states
>   - Elements can be navigated to and used by just a keyboard
>   - Elements and text can be read out by a screen reader, and the descriptions are understandable
> - Your feature / bug fix works across **GCP** and **AWS** (where appropriate)
>   - Are modifications to eq-publisher-v3 required?
>   - Are modifications to the Firestore data source required?

---

### What is the context of this PR?

**Tickets:** https://jira.ons.gov.uk/browse/EAR-2033 and https://jira.ons.gov.uk/browse/EAR-2035
**Publisher PR:** https://github.com/ONSdigital/eq-publisher-v3/pull/160

_This PR is for both EAR 2033 and EAR 2035_
Redesign section summary and section introduction on section page
Add section summary and section introduction page descriptions, and remove section page description

### How to review

**Section introduction**
**Check:**
- [ ] Section page has a toggle switch to enable section introduction
- [ ] Section introduction title, content and page description input boxes are only displayed when section introduction page toggle is enabled
- [ ] Section introduction title, content and page description can be updated by typing into the input fields
- [ ] When section introduction is enabled and either introduction title or introduction content has text, the section preview tab is enabled
- [ ] When section introduction is enabled and neither introduction title nor introduction content has text, the section preview tab is not enabled
- [ ] When section introduction is not enabled and either introduction title or introduction content has text, the section preview tab is not enabled
- [ ] Section introduction matches its design

**Section summary**
**Check:**
- [ ] Section page has a toggle switch to enable section summary
- [ ] Section summary page description is only displayed when section summary toggle is enabled
- [ ] Section summary page description can be updated by typing into the input field

**Migrations**
**Check:**
- [ ] The section `pageDescription` value defined in pre-existing questionnaires is converted to the new `sectionSummaryPageDescription` value
- [ ] The section introduction page toggle is enabled in pre-existing questionnaires if the questionnaire had either introduction title or introduction content text
- [ ] Section introduction page title and content have a validation error if they were `null` in a pre-existing questionnaire (this can be done by leaving introduction title and content empty, before any text has been entered to them previously)
- [ ] Section introduction page description has a validation error in pre-existing questionnaires

**Other**
- [ ] Check other pre-existing page description fields in Author have unchanged titles - their text content have changed to follow the prototype, however "Description and definitions" and "Page description" titles remain unchanged

### What to do after everything is green

1. - [ ] Bring this branch up-to-date with Origin/Master
2. - [ ] If there are a lot of commits or some are not helpful to read, squash them down
3. - [ ] Click the **Merge** button on this pull request
4. - [ ] Does this change mean the **Capability examples** questionnaire should be updated?
5. - [ ] Move the Jira ticket for this task into the next stage of the process

### CODE REVIEW

File: eq-author-api/schema/tests/section.test.js [hunk 1]
```diff
@@ -256,68 +256,162 @@ describe("section", () => {
...
-      it("should be valid if neither introduction title nor introduction content are populated", async () => {
-        const update = {
-          id: section.id,
-          title: section.title,
-          introductionTitle: "",
-          introductionContent: "",
-        };
-        const updatedSection = await updateSection(ctx, update);
+      describe("Section introduction", () => {
+        it("should be valid if introduction is not enabled and introduction title, introduction content and page title are populated", async () => {
```
Comment (Paul-Joel) [inline eq-author-api/schema/tests/section.test.js:260]: Should this not say 'are not populated'

File: eq-author-api/tests/utils/contextBuilder/section/moveSection.js [hunk 2]
```diff
@@ -77,8 +77,12 @@ const moveSectionMutation = `
       sections {
         id
         title
-        pageDescription
-        displayName
```
Comment (Paul-Joel) [inline eq-author-api/tests/utils/contextBuilder/section/moveSection.js:81]: Should this be removed?

File: eq-author-api/tests/utils/contextBuilder/section/updateSection.js [hunk 3]
```diff
@@ -5,12 +5,20 @@ const updateSectionMutation = `
     updateSection(input: $input) {
       id
       title
-      pageDescription
+      introductionEnabled
```
Comment (Paul-Joel) [inline eq-author-api/tests/utils/contextBuilder/section/updateSection.js:8]: These appear to be repeating

File: eq-author/src/App/section/Design/index.js [hunk 4]
```diff
@@ -191,7 +191,10 @@ export const UnwrappedSectionRoute = (props) => {
   };
 
   const hasIntroductionContent = () => {
-    return Boolean(section?.introductionTitle || section?.introductionContent);
+    return Boolean(
+      section?.introductionEnabled &&
+        (section?.introductionTitle || section?.introductionContent)
```
Comment (Paul-Joel) [inline eq-author/src/App/section/Design/index.js:196]: Should this not have an 'or' for introductionPageDescription as well?

File: eq-author/src/App/section/Design/index.test.js [hunk 5]
```diff
@@ -465,14 +468,15 @@ describe("SectionRoute", () => {
       });
     });
 
-    it("should enable the preview tab when the introduction is enabled", async () => {
+    it("should enable the preview tab when introduction is enabled and introduction title or content is not empty", async () => {
```
Comment (Paul-Joel) [inline eq-author/src/App/section/Design/index.test.js:471]: Should this not also be checking for introductionPageDescription? (and in other tests below)

**Review by Paul-Joel — CHANGES_REQUESTED** [review]
Couple of minor changes

File: eq-author-api/tests/utils/contextBuilder/section/moveSection.js (hunk 2 above)
Comment (farres1) [reply eq-author-api/tests/utils/contextBuilder/section/moveSection.js:81]: `pageDescription` - removed to be replaced with `sectionSummaryPageDescription`

`displayName` - will re-add this value

**Review by Paul-Joel — APPROVED** [review]

**Review by sudeepkunhis — APPROVED** [review]

**Pull request merged**